vault/eventbus/bus.go

```diff
@@ -247,11 +298,12 @@ func newFilterNode(ns *namespace.Namespace, pattern string) *eventlogger.Filter
 	}
 }
 
-func newAsyncNode(ctx context.Context, namespace *namespace.Namespace, logger hclog.Logger) *asyncChanNode {
+func newAsyncNode(ctx context.Context, logger hclog.Logger, filter *eventlogger.Filter) *asyncChanNode {
 	return &asyncChanNode{
 		ctx:    ctx,
 		ch:     make(chan *logical.EventReceived),
 		logger: logger,
+		filter: filter,
```

Comment:
This feels like the wrong model if we have to stuff all the nodes that precede our sink in the pipeline into the sink itself as well. Can we coordinate the replay outside of the sink node?

Reply:
I would definitely like to. I'll see if I can come up with a way to do that.